libutils/mustache.c

```diff
@@ -447,14 +457,54 @@ static bool RenderVariable(Buffer *out,
     switch (JsonGetElementType(var))
     {
     case JSON_ELEMENT_TYPE_PRIMITIVE:
-        // note that this also covers 'serialize' on primitives
-        return RenderVariablePrimitive(out, var, escape, key_mode);
+        if (key_mode && SeqLength(hash_stack) > 1)
+        {
+            JsonElement *parent = SeqAt(hash_stack, SeqLength(hash_stack) - 2);
+
+            if (JsonGetElementType(parent) == JSON_ELEMENT_TYPE_CONTAINER &&
+                JsonGetContainerType(parent) == JSON_CONTAINER_TYPE_ARRAY)
+            {
+                BufferAppendF(out, "%jd", iteration_position);
+                return true;
+            }
+            else
+            {
+                return RenderVariablePrimitive(out, var, escape, key_mode);
+            }
+        }
+        else
+        {
+            // note that this also covers 'serialize' on primitives
+            return RenderVariablePrimitive(out, var, escape, key_mode);
+        }
 
     case JSON_ELEMENT_TYPE_CONTAINER:
         if (serialize || serialize_compact)
         {
             return RenderVariableContainer(out, var, serialize_compact);
         }
+        else if (key_mode && SeqLength(hash_stack) > 1)
+        {
+            JsonElement *parent = SeqAt(hash_stack, SeqLength(hash_stack) - 2);
+
+            if (JsonGetElementType(parent) == JSON_ELEMENT_TYPE_CONTAINER &&
+                JsonGetContainerType(parent) == JSON_CONTAINER_TYPE_OBJECT)
+            {
+                // this will only extract the key property
+                return RenderVariablePrimitive(out, var, escape, key_mode);
+            }
+            else if (JsonGetElementType(parent) == JSON_ELEMENT_TYPE_CONTAINER &&
+                     JsonGetContainerType(parent) == JSON_CONTAINER_TYPE_ARRAY)
+            {
+                BufferAppendF(out, "%jd", iteration_position);
+                return true;
+            }
+            else
+            {
+                assert(false);
+                return false;
+            }
+        }
         else
         {
             assert(false);
```

Comment:
Same here, you can get rid of all else blocks, since all earlier blocks return and the remaining ones will end up at the assert and return below.